osu.Game/Overlays/Options/SliderOption.cs

```diff
@@ -1,53 +1,155 @@
+                        Height = 2,
+                        RelativeSizeAxes = Axes.None,
+                        Anchor = Anchor.CentreLeft,
+                        Origin = Anchor.CentreLeft,
+                        Colour = new Color4(255, 102, 170, 255),
+                    },
+                    rightBox = new Box
+                    {
+                        Height = 2,
+                        RelativeSizeAxes = Axes.None,
+                        Anchor = Anchor.CentreRight,
+                        Origin = Anchor.CentreRight,
+                        Colour = new Color4(255, 102, 170, 255),
+                        Alpha = 0.5f,
+                    },
+                    nub = new Container
+                    {
+                        Width = Height,
+                        Height = Height,
+                        CornerRadius = Height / 2,
+                        Origin = Anchor.TopCentre,
+                        AutoSizeAxes = Axes.None,
+                        RelativeSizeAxes = Axes.None,
+                        Masking = true,
+                        BorderColour = new Color4(255, 102, 170, 255),
+                        BorderThickness = 3,
+                        Children = new[]
+                        {
+                            new Box { Colour = Color4.Transparent, RelativeSizeAxes = Axes.Both }
+                        }
+                    },
+                };
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(AudioManager audio)
+            {
+                sample = audio.Sample.Get(@"Sliderbar/sliderbar");
+            }
+
+            private void playSample()
+            {
+                if (Clock == null)
+                    return;
+                if (Clock.CurrentTime - lastSample > 50)
+                {
+                    lastSample = Clock.CurrentTime;
+                    sample.Frequency.Value = 1 + NormalizedValue * 0.2f;
+                    sample.Play();
+                }
+            }
+            
+            protected override bool OnKeyDown(InputState state, KeyDownEventArgs args)
+            {
+                if (args.Key == Key.Left || args.Key == Key.Right)
+                    playSample();
+                return base.OnKeyDown(state, args);
+            }
+            
+            protected override bool OnClick(InputState state)
+            {
+                playSample();
+                return base.OnClick(state);
+            }
+            
+            protected override bool OnDrag(InputState state)
+            {
+                playSample();
+                return base.OnDrag(state);
+            }
+
+            protected override void UpdateValue(float value)
+            {
+                nub.MoveToX(DrawWidth * value, 300, EasingTypes.OutQuint);
+                leftBox.ScaleTo(new Vector2(
+                        MathHelper.Clamp(DrawWidth * value - nub.Width / 2 + 2, 0, DrawWidth),

```

Comment:
This clamp is the reason for the first issue I pointed out earlier. I recommend you only apply a transformation to the nub, and inside an overridden `Update` you update the scales of the right and left boxes based on the nub position.